hash.go

```diff
@@ -0,0 +1,31 @@
+package dns
+
+import (
+	"bytes"
+	"crypto"
+	"hash"
+)
+
+// identityHash will not hash, it only buffers the data written into it and returns it as-is.
+type identityHash struct {
```

Comment:
Rather than adding this, what about if instead we did something like:
```go
func hashFromAlgorithm(alg uint8, data ...[]byte) (hash.Hash, []byte, error)
```

Then just perform the hashing or appending inside that function?

I don't know if that's nicer or not, but it could be.

Reply:
I think that would be shorter (in the end), however there are some shenigans in the sig0 code, where there is a couple of lines of:
~~~
	sigend := offset
	h.Write(buf[sigstart:sigend])
	h.Write(buf[:10])
	h.Write([]byte{
		byte((adc - 1) << 8),
		byte(adc - 1),
	})
	h.Write(buf[12:bodyend])

	hashed := h.Sum(nil)
~~~
and the `hashFromAlgorithm` is some lines back.

So yes, probably a good idea to clean further, but we don't already have the correct bytes in a neat slice when we call this.